pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -123,15 +124,15 @@ func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, vmi *v1.
 	return nil
 }
 
-func readIPAddressesFromLink(podInterfaceName string) (string, string, error) {
-	link, err := Handler.LinkByName(podInterfaceName)
+func (l *podNICImpl) readIPAddressesFromLink(podInterfaceName string) (string, string, error) {
```

Comment:
I don't see any connection between this method and `podNICImpl`. It is just a helper method for some of the bindings.
I find it very strange that `podNICImpl` created the `BindingMechanism` and also has method used by the `BindingMechanism` it created.